Sources/Extensions/SwiftStdlib/DictionaryExtensions.swift

```diff
@@ -30,8 +30,8 @@ public extension Dictionary {
     ///		dict.keys.contains("key2") -> false
     ///
     /// - Parameter keys: keys to be removed
-    public mutating func removeAll(keys: [Key]) {
-        keys.forEach({ removeValue(forKey: $0)})
+    public mutating func removeAll<S: Sequence>(keys: S) where S.Element == Key {
+        Set<Key>(keys).forEach({ removeValue(forKey: $0) })
```

Comment:
Is it necessary to create the `Set` at all? I think this qualifies as [premature optimisation](http://wiki.c2.com/?PrematureOptimization).

Let's calculate:
`n` = size of `self`
`k1` = original size of `keys`
`k2` = size of `Set(keys)`
Creating the set costs `O(k1)` (at best).
According to the documentation, `removeValue(forKey:)` costs `O(n)`
Therefore, removing values costs `O(k2 * n)` (assuming no keys were actually removed, for arguments' sake)
So the total cost = `O(k1 + (k2 * n))`
There's a decent chance that callers of this method won't use repeated keys (but who knows), so it would be more like `O(2 * k1 * n)`

If you don't create the set then it would simply be `O(k1 * n)` (i.e. twice as fast). And if somebody finds in writing their app that this function takes a long time because of repeated keys then they could optimise it themselves by creating the set.

Reply:
I see the point of letting this decision to the user, I think it's o good option :))

Reply:
Btw nice article about premature optimisation, thank's for sharing 👍

Reply:
My old boss used to always call us up on it. He successfully drilled it into my head 🤯

Reply:
Good thoughts